source/renderer/app/components/loading/StatusIcons.js

```diff
@@ -0,0 +1,257 @@
+      'Message "Checking if Node time is correct" on the status icon tooltip',
+  },
+  nodeIsSyncing: {
+    id: 'status.icons.nodeIsSyncing',
+    defaultMessage: 'Node is syncing!',
+    description: 'Message "Node is syncing" on the status icon tooltip',
+  },
+  nodeIsNotSyncing: {
+    id: 'status.icons.nodeIsNotSyncing',
+    defaultMessage: 'Node is not syncing!',
+    description: 'Message "Node is not syncing" on the status icon tooltip',
+  },
+  checkingIfNodeIsSyncing: {
+    id: 'status.icons.checkingIfNodeIsSyncing',
+    defaultMessage: 'Checking if Node is syncing!',
+    description:
+      'Message "Checking if Node is syncing" on the status icon tooltip',
+  },
+});
+
+type Props = {
+  nodeState: ?CardanoNodeState,
+  isNodeResponding?: boolean,
+  isNodeSubscribed?: boolean,
+  isNodeTimeCorrect?: boolean,
+  isNodeSyncing?: boolean,
+};
+
+const STATUS_CLASSNAMES: Object = {
+  [CardanoNodeStates.STARTING]: 'unloaded',
+  [CardanoNodeStates.RUNNING]: 'on',
+  [CardanoNodeStates.EXITING]: 'unloaded',
+  [CardanoNodeStates.STOPPING]: 'unloaded',
+  [CardanoNodeStates.STOPPED]: 'unloaded',
+  [CardanoNodeStates.UPDATING]: 'unloaded',
+  [CardanoNodeStates.UPDATED]: 'unloaded',
+  [CardanoNodeStates.CRASHED]: 'off',
+  [CardanoNodeStates.ERRORED]: 'off',
+  [CardanoNodeStates.UNRECOVERABLE]: 'off',
+  true: 'on',
+  false: 'off',
+  undefined: 'unloaded',
+};
+
+const MESSAGES = {
+  [CardanoNodeStates.RUNNING]: messages.nodeIsRunning,
+  [CardanoNodeStates.STARTING]: messages.nodeIsStarting,
+  [CardanoNodeStates.EXITING]: messages.nodeIsExiting,
+  [CardanoNodeStates.STOPPING]: messages.nodeIsStopping,
+  [CardanoNodeStates.STOPPED]: messages.nodeHasStopped,
+  [CardanoNodeStates.UPDATING]: messages.nodeIsUpdating,
+  [CardanoNodeStates.UPDATED]: messages.nodeHasBeenUpdated,
+  [CardanoNodeStates.CRASHED]: messages.nodeHasCrashed,
+  [CardanoNodeStates.ERRORED]: messages.nodeHasErrored,
+  [CardanoNodeStates.UNRECOVERABLE]: messages.nodeIsUnrecoverable,
+  nodeStatenull: messages.nodeIsStarting,
+  isNodeRespondingtrue: messages.nodeIsResponding,
+  isNodeRespondingfalse: messages.nodeIsNotResponding,
+  isNodeRespondingundefined: messages.checkingIfNodeIsResponding,
+  isNodeSubscribedtrue: messages.nodeIsSubscribed,
+  isNodeSubscribedfalse: messages.nodeIsNotSubscribed,
+  isNodeSubscribednull: messages.checkYourInternetConnection,
+  isNodeSubscribedundefined: messages.checkingIfNodeIsSubscribed,
+  isNodeTimeCorrecttrue: messages.nodeTimeIsCorrect,
+  isNodeTimeCorrectfalse: messages.nodeTimeIsNotCorrect,
+  isNodeTimeCorrectundefined: messages.checkingIfNodeTimeIsCorrect,
+  isNodeSyncingtrue: messages.nodeIsSyncing,
+  isNodeSyncingfalse: messages.nodeIsNotSyncing,
+  isNodeSyncingundefined: messages.checkingIfNodeIsSyncing,
+};
+
+export default class StatusIcon extends Component<Props> {
+  static contextTypes = {
+    intl: intlShape.isRequired,
+  };
+
+  getTip = (paramName: string, paramValue?: boolean) => {
```

Comment:
from what i see `paramValue` can be either `boolean`, `string`, `null` or `undefined` right? If so, it would be good to indicate that in the signature like `paramValue?: boolean | string | null | undefined` (and maybe extract that into a separate named type)

Reply:
Done